### ISSUE TITLE
exitcode 1 on agents when using kubernetes backend

### ISSUE THREAD

**Issue by jaqua** [woodpecker-ci/woodpecker#1696]
### Component

server, agent

### Describe the bug

I installed woodpecker next version on my kubernetes cluster in `ci` namespace and connect it to gitea. So far it is working, but I cannot run any pipeline. My agents give me the error:

```
{"level":"info","time":"2023-04-02T21:43:10Z","message":"Starting Woodpecker agent with version 'next-ea95d5aa' and backend 'kubernetes' using platform 'linux/amd64' running up to 1 pipelines in parallel"}
{"level":"error","ID":"1","Repo":"project/source","Pipeline":"1","error":"pods \"wp-01gx1yvm3dw62np4dmpjbqkq16-0-clone-0\" is forbidden: User \"system:serviceaccount:ci:woodpecker-agent\" cannot delete resource \"pods\" in API group \"\" in the namespace \"ci\"","time":"2023-04-02T21:44:31Z","message":"could not destroy engine"}
{"level":"error","error":"rpc error: code = Unknown desc = Step finished with exitcode 1, persistentvolumeclaims is forbidden: User \"system:serviceaccount:ci:woodpecker-agent\" cannot create resource \"persistentvolumeclaims\" in API group \"\" in the namespace \"ci\"","time":"2023-04-02T21:44:31Z","message":"grpc error: wait(): code: Unknown: rpc error: code = Unknown desc = Step finished with exitcode 1, persistentvolumeclaims is forbidden: User \"system:serviceaccount:ci:woodpecker-agent\" cannot create resource \"persistentvolumeclaims\" in API group \"\" in the namespace \"ci\""}
{"level":"warn","repo":"project/source","pipeline":"1","id":"1","error":"rpc error: code = Unknown desc = Step finished with exitcode 1, persistentvolumeclaims is forbidden: User \"system:serviceaccount:ci:woodpecker-agent\" cannot create resource \"persistentvolumeclaims\" in API group \"\" in the namespace \"ci\"","time":"2023-04-02T21:44:31Z","message":"cancel signal received"}
```

These are the values for my helm charts to install server and agent:

**woodpecker/woodpecker-server**
```
image:
  tag: "next"
persistentVolume:
  storageClass: longhorn
extraSecretNamesForEnvFrom:
  - woodpecker-secret
env:
  WOODPECKER_GITHUB: false
  WOODPECKER_GITEA: true
  WOODPECKER_GITEA_URL: https://git.domain.com
  WOODPECKER_HOST: https://ci.domain.com
```

**woodpecker/woodpecker-agent**
```
image:
  tag: "next"
extraSecretNamesForEnvFrom:
  - woodpecker-secret
env:
  WOODPECKER_SERVER: woodpecker-server.ci.svc.cluster.local:9000
  WOODPECKER_BACKEND: kubernetes
```

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"next-ea95d5aa"}
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by zc-devs** [comment]
Woodpecker agent needs permissions to manage `pods`, `logs` and and `persistentvolumeclaims`. You should have service account `woodpecker-agent`  already and have to create `Role` and bind it to service account.

Take a look to my config. I have Gitea, Woodpecker's server and agent in `scm` namespace, but run pipelines in `woodpecker-runtime` namespace. If you want to run pipelines in the same namespace, replace all namespaces to your `ci` value.

```yaml
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: woodpecker-agent
  namespace: scm
  labels:
    app.kubernetes.io/name: woodpecker-agent
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  namespace: woodpecker-runtime
  name: woodpecker-agent
rules:
  - apiGroups: [""]
    resources: ["persistentvolumeclaims", "pods", "pods/log"]
    verbs: ["*"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: woodpecker-agent
  namespace: woodpecker-runtime
subjects:
  - kind: ServiceAccount
    name: woodpecker-agent
    namespace: scm
roleRef:
  kind: Role
  name: woodpecker-agent
  apiGroup: rbac.authorization.k8s.io
```

**Comment by pat-s** [comment]
@jaqua If this is still an issue, please open a new issue in https://github.com/woodpecker-ci/helm
